### PR TITLE
[FIX/ENH] Improve generate_report method to work with fit_transform for the maskers

### LINKED ISSUE #3709: [ENH] generate_report method should also work after fit_transform was called
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed enhancement in detail.

Currently, it seems to be the case that the `.generate_report` method of masker objects can only be called with a previous call of `.fit()` but not when `.fit_transform()` was called (which is contra-intuitive to me because `.fit_transform()` is just `.fit()` + `.transform()`). Most of the time, users will probably use `.fit_transform()` to get their numpy arrays. If they then want to check if the masking worked they currently have to call `.fit()` one more time to be able to call  `.generate_report`.

### Benefits to the change

Avoids calling `masker.fit()` one more time if a report should be generated. 

### Pseudocode for the new behavior, if applicable

MRE:

```python
from nilearn import datasets
from nilearn.maskers import NiftiLabelsMasker
import numpy as np
from nilearn.image import load_img

# download a single statistical image
motor_images = datasets.fetch_neurovault_motor_task()
stat_img = motor_images.images[0]

# get an atlas
atlas_data = datasets.fetch_atlas_schaefer_2018()

# get the atlas image. This has 401 values, with the 0 included
atlas_img = load_img(atlas_data['maps'])
print(len(np.unique(atlas_img.get_fdata())))

# we have to insert a 0 here so that img and labels have the same number of value
# This is already reported, see #3708
atlas_labels = atlas_data.labels.tolist()
atlas_labels.insert(0,'Background') 

# set a masker
masker = NiftiLabelsMasker(labels_img=atlas_img,labels=atlas_labels)

```
**Would be nice if this would work!**

```python

data = masker.fit_transform(stat_img)
report = masker.generate_report()
report.open_in_browser()
```

**But we have to call .fit() one more time in order to be able to generate a report**

```
masker.fit(stat_img)
report = masker.generate_report()
report.open_in_browser()
```

### PR DESCRIPTION
**Pull request by ymzayek**
- Deals with most of #3709

Changes proposed in this pull request:

- [x] Pass image data to fit() in fit_transform()
- [x] Use better colormap or instead allow users to pass colormap
- [x] Improve docstrings to explain image data being passed to reporter
- [ ] Store 1 time point for reporting using 4D images (will handle in a separate PR)

### CODE REVIEW

**Comment by ymzayek** [comment]
Now 
```python
data = masker.fit_transform(img)
report = masker.generate_report()
```

works the same as:
```python
masker.fit(img)
report = masker.generate_report()
```
to generate: 
Untitled


I changed the colormap but maybe there are suggestions for something better??

**Comment by ymzayek** [comment]
Storing one time point for the report seems a bit more involved than I thought given the failures here: https://github.com/nilearn/nilearn/actions/runs/5787425257 so I will handle it in a separate follow-up PR

**Comment by bthirion** [comment]
> 

The red background is not great for the atlas display. I'll try to get a neutral one.

**Review by bthirion — COMMENTED** [review]
LGTM otherwise

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Review by Remi-Gau — APPROVED** [review]
Good with me!
